app/src/main/java/tttTDD/Java/GameInput.java

```diff
@@ -0,0 +1,50 @@
+package tttTDD.Java;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class GameInput {
+
+    public static int playerSelection(Scanner input, Display display) {
+        int value;
+
+        try {
+            value = input.nextInt();
+            if (value > 0 || value < 4) {
+                return value;
+            }
+        }
+
+        catch (Exception e) {
+            System.out.println(display.wrongTypeMessage());
+            input.nextLine();
+            return  playerSelection(input, display);
+        }
+
+        return value;
+    }
+
+    public static int boardSize(Scanner input, Display display) {
+        try {
+            int value = input.nextInt();
+
+            if (value > 2) {
+                return value;
```

Comment:
You're still enforcing this constraint in two separate places (Board and GameInput), and with this refactor they are even more separate from each other. I think that the Board class is better suited for this check than this one, but it certainly shouldn't be both.

If I'm not mistaken, the program will currently fail when you specify a value of 2, because it will happily exit this recursive input method but then throw an error when attempting to initialize the Board.